sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClient.cs

```diff
@@ -362,5 +362,193 @@ public virtual AnalyzeDocumentOperation AnalyzeDocumentFromUri(WaitUntil waitUnt
                 throw;
             }
         }
+
+        /// <summary>
+        /// Classifies one or more documents using a document classifier built with custom documents.
+        /// </summary>
+        /// <param name="waitUntil">
+        /// <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service;
+        /// <see cref="WaitUntil.Started"/> if it should return after starting the operation.
+        /// </param>
+        /// <param name="classifierId">The ID of the document classifier to use.</param>
+        /// <param name="document">The stream containing one or more documents to classify.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>
+        /// A <see cref="ClassifyDocumentOperation"/> to wait on this long-running operation. Its <see cref="ClassifyDocumentOperation.Value"/> upon successful
+        /// completion will contain documents classified from the input.
+        /// </returns>
+        public virtual async Task<ClassifyDocumentOperation> ClassifyDocumentAsync(WaitUntil waitUntil, string classifierId, Stream document, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(classifierId, nameof(classifierId));
+            Argument.AssertNotNull(document, nameof(document));
+
+            using DiagnosticScope scope = Diagnostics.CreateScope($"{nameof(DocumentAnalysisClient)}.{nameof(ClassifyDocument)}");
+            scope.Start();
+
+            try
+            {
+                var response = await ServiceClient.DocumentClassifiersClassifyDocumentAsync(
+                    classifierId,
+                    InternalContentType.ApplicationOctetStream,
+                    Constants.DefaultStringIndexType,
+                    document,
+                    cancellationToken).ConfigureAwait(false);
+
+                var operation = new ClassifyDocumentOperation(ServiceClient, Diagnostics, response.Headers.OperationLocation, response.GetRawResponse());
+
+                if (waitUntil == WaitUntil.Completed)
```

Comment:
Nope. The operation has already started at this point. Since most customers will just call `WaitForCompletionAsync` right after starting the operation, we can do that for them when they pass in `WaitUntil.Completed`.

Reply:
I see. Does this mean that Form Recognizer never uses the "NotStarted" operation status?

Reply:
That's an interesting question. I haven't thought about that before but we do have a `NotStarted` operation status. This, however, has not relation to the [`WaitUntil` SDK pattern](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-longrunning). I agree that the naming can be misleading, though.

This means that it's possible to have an operation with `NotStarted` status after we created it by passing `WaitUntil.Started` to the service method.